src/plugin.js

```diff
@@ -232,19 +232,27 @@ class PlaylistMenu extends Component {
       this.addClass('vjs-mouse');
     }
 
-    player.on(['loadstart', 'playlistchange', 'playlistsorted'], (event) => {
+    this.on(player, ['loadstart', 'playlistchange', 'playlistsorted'], (event) => {
       this.update();
     });
 
     // Keep track of whether an ad is playing so that the menu
     // appearance can be adapted appropriately
-    player.on('adstart', () => {
+    this.on(player, 'adstart', () => {
       this.addClass('vjs-ad-playing');
     });
 
-    player.on('adend', () => {
+    this.on(player, 'adend', () => {
       this.removeClass('vjs-ad-playing');
     });
+
+    this.on('dispose', () => {
+      player.playlistMenu = null;
+    });
+
+    this.on(player, 'dispose', () => {
+      this.dispose();
```

Comment:
looks like PlaylistMenuItems are created and added manually rather than using the video.js component API, so, we need to dispose them manually as well
https://github.com/brightcove/videojs-playlist-ui/blob/767b7ac4d3f5986f07a9b49f43c3c56202fa754c/src/plugin.js#L272-L279

Reply:
Bahhhhhhhh.